transcom/mymove
# [MB-16182] Auth added to update mto shipments endpoint

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16182)

## Summary

This PR adds an authz check for the POST `/mto-shipments` endpoint. It checks that the service member associated with the session matches the service member associated with the move that the shipment will be created for

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Start server and customer client.
2. Log in as a service member that has just submitted their move
3. Look at the network tab and find the value for the `MoveTaskOrderID`
4. Log out and log in as another service member to get their cookie info for postman. Load into postman based on however you set up postman.
    1. [Docs for setting up postman](https://transcom.github.io/mymove-docs/docs/tools/postman/setting-up-postman)
5. Make a patch request in postman to `/mto-shipments` 
    1. Since this is going to be a patcht request, you'll need to [set the `X-CSRF-TOKEN` header to be the `masked_gorilla_csrf` cookie value](https://transcom.github.io/mymove-docs/docs/tools/postman/intercepting-cookies-for-postman#form-a-working-get-and-patch-request-in-postman).
    2. You will also need a valid body to send along with the request. I used the following and note the `ShipmentID` was from the non logged in user (not the user I retrieved the cookies for)
```
{
    "estimatedWeight": 4200,
    "shipmentType": "HHG"
}
```
6. Verify you get a 404 not found response.
  
<img width="950" alt="Screenshot 2023-07-26 at 11 19 11 AM" src="https://github.com/transcom/mymove/assets/67110378/e65a3fc8-4581-4469-9a41-f8770c24eb19">
